src/library/scala/jdk/javaapi/FutureConverters.scala

```diff
@@ -72,7 +72,12 @@ object FutureConverters {
       case f: Future[T @unchecked] => f
       case _ =>
         val p = new P[T](cs)
-        cs whenComplete p
+        val future = cs.toCompletableFuture
+        if (future.isDone && !future.isCompletedExceptionally) {
+          p.tryComplete(Success(future.join()))
+        } else {
+          cs.handle(p)
```

Comment:
Do you know why `handle` is faster than `whenComplete`? Is the performance change significant?

Reply:
Expecially on the failed case, I think the main outcome is from avoiding the `encodeThrowable`